include/pybind11/cast.h

```diff
@@ -435,12 +435,21 @@ template <typename StringType, bool IsView = false> struct string_caster {
                 return true;
             }
         }
+        if (PyByteArray_Check(src.ptr())) {
+            // We were passed a bytearray; accept it into a std::string or char*
+            // without any encoding attempt.
+            const char *bytes = PyByteArray_AsString(src.ptr());
+            if (bytes) {
```

Comment:
I'm uncertain if there will be a case bytes is nullptr/NULL. I agree with you we better shout out for such an error. 
I'll wait for more comments for this one.

Reply:
I see you added the checks, thank you!

This PR looks great to me. The only reason I'm holding off clicking the approve button is that I want to run this through our (Google's) global testing. It will probably take me a couple days to restore my ability to do that; I need to merge master into the smart_holder branch first, which will take some effort after the sweeping clang-format changes.

I'll get back here asap.